### ISSUE TITLE
spaces_inside_linter() doesn't allow spacing for aligned indentation

### ISSUE THREAD

**Issue by MichaelChirico** [r-lib/lintr#1979]
```r
foo(
  c( 45, 90, 135),
  c(135, 90,  45)
)
````

Consider `c( 45,`, if we try to induce the alignment in this way, it throws `spaces_inside_linter()`, but if we try to induce it as `c(45 ,`, it throws `commas_linter()` instead.

Given that `infix_spaces_linter()` has `allow_multiple_spaces=TRUE` by default, this creates some inconsistency where most of the code allows spacing-for-alignment, except for this one initial entry.

**Comment by AshesITR** [comment]
IINM left-alignment is possible like this:

```r
foo(
  c(45,  90, 135),
  c(135, 90, 45 )
)
```

**Comment by MichaelChirico** [comment]
that's right, but it looks odd to have that one comma out of place when the rest of the lines are aligned by commas. it's more obvious with more entries:

```r
foo(
  c(45,  90, 135,  45, 90, 135),
  c(135, 90,  45, 135, 90,  45)
)
```

what I've actually done for now is 0-pad that entry:

```r
foo(
  c(045, 90, 135,  45, 90, 135),
  c(135, 90,  45, 135, 90,  45)
)
```

**Comment by AshesITR** [comment]
I agree that right-aligned code in your example would be nicer to read.
I'm wondering how we could reliably detect such a use of space for alignment.
Some examples to look at:

```r
f( 42 )
f(  5 )

f(42  )
f( 0.5)

f( 3, 42)
f( 5, 67)
f(89,  1)
```

**Comment by MichaelChirico** [comment]
The logic will definitely be wayyyy more complicated than currently. {styler} has an associated vignette:

https://cran.r-project.org/web/packages/styler/vignettes/detect-alignment.html

This is definitely a nice-to-have, so I changed labels to 'feature' instead of 'false-positive'.
